flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/catalog/MySqlCatalogTestBase.java

```diff
@@ -111,39 +126,338 @@ class MySqlCatalogTestBase {
+                            Time.valueOf("09:33:43").toLocalTime(),
+                            Timestamp.valueOf("2021-08-04 01:53:19.098").toLocalDateTime(),
+                            null));
+
+    private MySqlCatalog catalog;
+    private TableEnvironment tEnv;
+
+    protected static MySQLContainer<?> createContainer(String dockerImage) {
+        return new MySQLContainer<>(DockerImageName.parse(dockerImage))
+                .withUsername("root")
+                .withPassword("")
+                .withEnv(DEFAULT_CONTAINER_ENV_MAP)
+                .withInitScript(MYSQL_INIT_SCRIPT)
+                .withLogConsumer(new Slf4jLogConsumer(LOG));
+    }
+
+    protected abstract String getDatabaseUrl();
+
+    @BeforeEach
+    void setup() {
+        catalog =
+                new MySqlCatalog(
+                        Thread.currentThread().getContextClassLoader(),
+                        TEST_CATALOG_NAME,
+                        TEST_DB,
+                        TEST_USERNAME,
+                        TEST_PWD,
+                        getDatabaseUrl());
+
+        this.tEnv = TableEnvironment.create(EnvironmentSettings.inStreamingMode());
+        tEnv.getConfig().set(TABLE_EXEC_RESOURCE_DEFAULT_PARALLELISM, 1);
+
+        // Use mysql catalog.
+        tEnv.registerCatalog(TEST_CATALOG_NAME, catalog);
+        tEnv.useCatalog(TEST_CATALOG_NAME);
+    }
+
+    @Test
+    void testGetDb_DatabaseNotExistException() throws Exception {
+        String databaseNotExist = "nonexistent";
+        assertThatThrownBy(() -> catalog.getDatabase(databaseNotExist))
+                .satisfies(
+                        anyCauseMatches(
+                                DatabaseNotExistException.class,
+                                String.format(
+                                        "Database %s does not exist in Catalog",
+                                        databaseNotExist)));
+    }
+
+    @Test
+    void testListDatabases() {
+        List<String> actual = catalog.listDatabases();
+        assertThat(actual).containsExactly(TEST_DB, TEST_DB2);
+    }
+
+    @Test
+    void testDbExists() throws Exception {
+        String databaseNotExist = "nonexistent";
+        assertThat(catalog.databaseExists(databaseNotExist)).isFalse();
+        assertThat(catalog.databaseExists(TEST_DB)).isTrue();
+    }
+
+    // ------ tables ------
+
+    @Test
+    void testListTables() throws DatabaseNotExistException {
+        List<String> actual = catalog.listTables(TEST_DB);
+        assertThat(actual)
+                .isEqualTo(
+                        Arrays.asList(
+                                TEST_TABLE_ALL_TYPES,
+                                TEST_SINK_TABLE_ALL_TYPES,
+                                TEST_TABLE_SINK_FROM_GROUPED_BY,
+                                TEST_TABLE_PK));
+    }
+
+    @Test
+    void testListTables_DatabaseNotExistException() throws DatabaseNotExistException {
```

Comment:
```suggestion
    void testListTables_DatabaseNotExistException() {
```
exception from signature could be removed since it's handled in `assertThatThrownBy`

Reply:
this code was only moved here from MySqlCatalogITCase, but I can change it..

Reply:
done